tests/test_ops_qubit.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the available built-in discrete-variable quantum operations.
+"""
+from functools import reduce
+
+import pytest
+import numpy as np
+
+from pennylane.ops import qubit
+from pennylane.plugins.default_qubit import Rot3
+
+
+class TestHadamard:
+    """Test functions for Hadamard class."""
+    def test_hadamard_matrix(self):
+        """Test the Hadamard matrix representation"""
+        hadamard = qubit.Hadamard(0)
+        matrix = hadamard.matrix()
+        true_matrix = np.array([[1, 1], [1, -1]], dtype=complex) / np.sqrt(2)
+
+        assert np.allclose(matrix, true_matrix)
+
+    def test_hadamard_diagonalization(self):
+        """Test the Hadamard diagonalizing_gates function; returned operations
+        should transform the Hadamard gate into the Z-gate.
+
+        Only for testing diagonalization based on Rot().
+        """
+        hadamard = qubit.Hadamard(0)
+        zgate = np.array([[1, 0], [0, -1]], dtype=complex)
+
+        gate_list = []
+        for op in hadamard.diagonalizing_gates():
+            params = op.__dict__["params"]
```

Comment:
It should also work using the `op.params` property to extract the parameters from the operation

Reply:
Thanks! That's much better.